src/android/FileTransfer.java

```diff
@@ -316,7 +316,7 @@ public void run() {
                     conn.setRequestProperty("Content-Type", "multipart/form-data; boundary=" + BOUNDARY);
 
                     // Set the cookies on the response
-                    String cookie = CookieManager.getInstance().getCookie(target);
+                    String cookie = webView.getCookieManager().getCookie(target);
```

Comment:
I think we'll need to use reflection here to keep the plugin compatible with 3.x versions of android.

e.g.: Look at how getPluginManager() is handled on line 854.